src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusCollectionManager.cs

```diff
@@ -262,6 +270,40 @@ private ExportResult OnCollect(Batch<Metric> metrics)
         }
     }
 
+    private int WriteTargetInfo()
+    {
+        if (this.targetInfoBuffer == null)
+        {
+            while (true)
+            {
+                try
+                {
+                    var cursor = PrometheusSerializer.WriteTargetInfo(this.buffer, 0, this.exporter.Resource);
+
+                    this.targetInfoBuffer = new byte[cursor];
+                    Array.Copy(this.buffer, 0, this.targetInfoBuffer, 0, cursor);
+
+                    break;
+                }
+                catch (IndexOutOfRangeException)
+                {
+                    if (!this.IncreaseBufferSize())
+                    {
+                        throw;
+                    }
+                }
+            }
+        }
+        else
+        {
+            // Buffer will always be large enough for the target_info at this point because we've already
+            // previously written to the buffer (increasing it as needed) and copied out the target_info buffer.
+            this.targetInfoBuffer.CopyTo(this.buffer, 0);
```

Comment:
Instead of saving and copying the targetinfo each time, we could adjust the base cursor position from 0 to (i+1)th index.  ith index would be the last byte of targetinfo.

Reply:
Good thinking 99, done.